Added missing import in boxplot_frame_groupby example

gsyqax opened the pull request:
- [X] closes #28903 
- [x] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

gsyqax committed:
Added missing import in boxplot_frame_groupby example

MarcoGorelli commented:
Thanks @gsyqax - did you build this page of the documentation locally to check it works?

gsyqax commented:
@MarcoGorelli yes, it is working.

I noticed the example also makes use of 'pd' and 'np' alias without importing and aliasing pandas and numpy. I don't think this can lead to confusing, but do you think I should also add those imports?

MarcoGorelli commented:
That's OK, it fits with the [docstring guide](https://pandas.pydata.org/docs/development/contributing_docstring.html)

> Import required libraries (except numpy and pandas)

MarcoGorelli commented:
> @MarcoGorelli yes, it is working.

I just tried building it with your changes and I don't see any plot though, do you?

gsyqax commented:
I'm not seeing any plots either. I thought a plot was not needed since the current version is not showing any either.

I will add the plots and commit again

gsyqax committed:
Added plot in boxplot_frame_groupby example

pep8speaks commented:
Hello @gsyqax! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/plotting/_core.py`](https://github.com/pandas-dev/pandas/blob/855448d3ba4b45c64cb366a3a2c22de586d3e60e/pandas/plotting/_core.py):

> [Line 554:1](https://github.com/pandas-dev/pandas/blob/855448d3ba4b45c64cb366a3a2c22de586d3e60e/pandas/plotting/_core.py#L554): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

gsyqax commented:
@MarcoGorelli I added some plots and made a few minor changes to make everything clearer, hope it's OK. Also removed whitespace as indicated by pep8speaks.

MarcoGorelli commented on pandas/plotting/_core.py:
```diff
@@ -547,17 +547,25 @@ def boxplot_frame_groupby(
+        >>> df = pd.DataFrame(data, columns=list('ABCD'), index=index)
+        >>> grouped = df.groupby(level='lvl1')
+        >>> boxplot_frame_groupby(grouped, rot=45, fontsize=12)
+        The ``subplots=False`` option shows the boxplots in a single figure.
```
```suggestion

    The ``subplots=False`` option shows the boxplots in a single figure.
```

MarcoGorelli reviewed (changes requested):
Nice! Thanks, just tried running it and it looks good. Can you just add a space between the two examples, as suggested below?

MarcoGorelli commented:
Also, I don't think a whatsnew note is necessary for documentation improvements, could you please revert that change?

gsyqax committed:
Update pandas/plotting/_core.py

Co-authored-by: Marco Gorelli <33491632+MarcoGorelli@users.noreply.github.com>

MarcoGorelli reviewed (approved):
LGTM

have opened #34353 as I'm not sure what to do about the subplots being too close (but it would warrant a separate PR anyway, I don't think there'd be any need to change that here)

gsyqax commented:
Great! Many thanks @MarcoGorelli for all the help, really appreciate it!

jreback commented:
cc @datapythonista @jorisvandenbossche @WillAyd

charlesdong1991 commented:
try `boxplot_frame_groupby(grouped, rot=45, fontsize=12, figsize=(8, 10))` for instance? @gsyqax 

it will avoid having overlapped subplots and make doc look better

jorisvandenbossche reviewed (commented):
IMO we should not advertise `boxplot_frame_groupby` like that directly, but rather as `df.groupby(..).boxplot(..)`

WillAyd commented:
Slight preference for what @jorisvandenbossche has suggested above but not a blocker so OK either way

gsyqax committed:
Beautified plots, changed boxplot_frame_groupby by groupby.boxplot method

gsyqax commented:
I applied the change to the plot fig size and replace boxplot_frame_groupby by df.groupby.boxplot. I'm keeping the intermediate var 'grouped' to make things easier to read

TomAugspurger commented on pandas/plotting/_core.py:
```diff
@@ -547,17 +547,26 @@ def boxplot_frame_groupby(
+        :context: close-figs
+
+        >>> import itertools
+        >>> from pandas.plotting import boxplot_frame_groupby
```
@gsyqax can you remove this import.

TomAugspurger replied:
```suggestion
```

TomAugspurger committed:
Update pandas/plotting/_core.py

jreback commented:
lgtm. @TomAugspurger

TomAugspurger commented:
Thanks @gsyqax!
